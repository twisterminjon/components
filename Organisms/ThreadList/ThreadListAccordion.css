.messagelistaccordion {
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: nowrap;

  overflow: hidden;
  height: 60px;
  width: 100%;

  padding-left: 16px;

  background-color: var(--messagepage_bg);

  border-bottom: solid 1px var(--brandcolor);
  text-decoration: none;
  cursor: default; /* don't show a text cursor */
  transition: background-color 250ms ease-in-out, transform 150ms ease;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}

.messagelistaccordion span {
  flex: 1;

  color: var(--button_color_hover);

  text-align: left;
  font-family: var(--font-message-page);
  font-size: var(--font-base-body--size);
  font-weight: 500;

  padding: 8px 12px;

  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.messagelistaccordion-button {
  display: flex;
  justify-content: center;
  align-items: center;

  background-color: var(--messagepage_bg);

  height: 60px;
  width: 60px;
}

.messagelistaccordion-button:hover {
  background-color: var(--button_color_hover);
}

.messagelistaccordion-button:focus {
  background-color: var(--button_color_hover);
}

.messagelistaccordion-button:active {
  transform: scale(1.01);
}

.messagelistaccordion-content {
  overflow: hidden;
  padding-bottom: 20px;
}

.messagelistaccordion--hide {
  height: 0;
  padding-bottom: 0;
}

.messagelistaccordion .button-custom {
  font-size: 16px !important;
}
