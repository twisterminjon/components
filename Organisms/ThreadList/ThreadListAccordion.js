import React from 'react'
import PropTypes from 'prop-types'

import './ThreadListAccordion.css'
import Button from '../../Atoms/Button/Button'

ThreadListAccordion.propTypes = {
  /** A name for the control, will be included with on onActivate property */
  name: PropTypes.string.isRequired,

  /** The label for the accordion */
  label: PropTypes.string.isRequired,

  /** Function to create a new Group */
  onClickGroup: PropTypes.func.isRequired,

  /** Primary content. */
  children: PropTypes.node,

  /** ClassName for the wrapper */
  className: PropTypes.string,

  /** Style for the wrapper */
  style: PropTypes.object,
}
ThreadListAccordion.defaultProps = {
  renderControls: null,
  className: '',
  style: {},
}

export default function ThreadListAccordion({ name, label, children, onClickGroup, className = '', style }) {
  return (
    <div className="messagelistaccordion-wrap">
      <div
        className={`messagelistaccordion ${className}`.trim()}
        style={style}
        data-testid={`message-list-accordion-${name}`}>
        <span>{label}</span>
        <Button
          data-testid="message-list-add-group"
          size="medium"
          style={{ fontSize: '12px !important' }}
          onClick={onClickGroup}>
          Add Group
        </Button>
      </div>
      {children && (
        <div className={`messagelistaccordion-content`.trim()} data-testid={`message-list-${name}-content`}>
          {children}
        </div>
      )}
    </div>
  )
}
