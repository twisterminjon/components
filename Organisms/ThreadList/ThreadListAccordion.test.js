import React from 'react'

import ThreadListAccordion from './ThreadListAccordion'

describe('ThreadListAccordion', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(
      <ThreadListAccordion label="polka" name="polka" onClickGroup={mockFun}>
        <p>test</p>
      </ThreadListAccordion>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
