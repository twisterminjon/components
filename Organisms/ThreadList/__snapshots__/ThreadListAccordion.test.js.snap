// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadListAccordion matches the snapshot 1`] = `
<div
  className="messagelistaccordion-wrap"
>
  <div
    className="messagelistaccordion"
    data-testid="message-list-accordion-polka"
    style={Object {}}
  >
    <span>
      polka
    </span>
    <Button
      data-testid="message-list-add-group"
      fluid={false}
      onClick={[MockFunction]}
      size="medium"
      style={
        Object {
          "fontSize": "12px !important",
        }
      }
      type="submit"
    >
      Add Group
    </Button>
  </div>
  <div
    className="messagelistaccordion-content"
    data-testid="message-list-polka-content"
  >
    <p>
      test
    </p>
  </div>
</div>
`;
